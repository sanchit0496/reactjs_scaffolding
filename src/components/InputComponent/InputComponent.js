import React from 'react'

const InputComponent = ({
  label,
  id,
  name,
  type,
  value,
  onChange,
  onBlur,
  error,
}) => {
  return (
    <div>
      <label htmlFor={id}>{label}</label>
      <input
        id={id}
        name={name}
        type={type}
        value={value}
        onChange={onChange}
        onBlur={onBlur}
      />
      {error && <div>{error}</div>}
    </div>
  )
}

export default InputComponent
