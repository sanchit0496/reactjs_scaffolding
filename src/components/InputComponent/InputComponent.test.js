import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import InputComponent from './InputComponent' // Adjust the path as necessary

describe('InputComponent', () => {
  // Snapshot Test
  it('renders correctly', () => {
    const { asFragment } = render(
      <InputComponent label="Test Label" id="test" name="test" type="text" />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  // Functional Test - Rendering and Props
  it('displays the correct label and input type', () => {
    render(
      <InputComponent label="Test Label" id="test" name="test" type="text" />,
    )
    expect(screen.getByLabelText('Test Label')).toBeInTheDocument()
    expect(screen.getByLabelText('Test Label')).toHaveAttribute('type', 'text')
  })

  // Functional Test - onChange Event
  it('calls onChange event when input changes', () => {
    const handleChange = jest.fn()
    render(
      <InputComponent
        label="Test Label"
        id="test"
        name="test"
        type="text"
        onChange={handleChange}
      />,
    )
    fireEvent.change(screen.getByLabelText('Test Label'), {
      target: { value: 'new value' },
    })
    expect(handleChange).toHaveBeenCalledTimes(1)
  })

  // Functional Test - Error Display
  it('displays an error message when error prop is provided', () => {
    const errorMessage = 'Error message'
    render(
      <InputComponent
        label="Test Label"
        id="test"
        name="test"
        type="text"
        error={errorMessage}
      />,
    )
    expect(screen.getByText(errorMessage)).toBeInTheDocument()
  })
})
