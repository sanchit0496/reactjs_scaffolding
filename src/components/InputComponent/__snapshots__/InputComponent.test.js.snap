// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent renders correctly 1`] = `
<DocumentFragment>
  <div>
    <label
      for="test"
    >
      Test Label
    </label>
    <input
      id="test"
      name="test"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
